### PR TITLE
New wizard used everywhere and some small fixes

### PR DESCRIPTION
**Pull request by denektenina**
# Description

Link to Jira issue: [https://sikt.atlassian.net/browse/NP-47431](https://sikt.atlassian.net/browse/NP-47431)

**⚠This is only visible when you have beta toggled to true in local storage ⚠**

This is a part of the effort to create a new create project-wizard [(NP-47281)](https://sikt.atlassian.net/browse/NP-47281), and there is a subtask for all the functionality to add.

In this PR we are ensuring the new wizard opens correctly wherever you try to access the new project or edit project view. You still have to have the beta flag toggled on to see the new view.

**The places that now open the new view are:**
* My page -> new project in the left sidepanel
* My page -> project list -> edit icon
* My page -> project list -> click project title to get to landing page -> edit icon
* New / edit registration -> first tab -> "Prosjekttilknytning"-field -> "Opprett nytt prosjekt": This will open the new project wizard in a new tab.

In all these views you should be able to create or edit your project. If you click save it should be persisted, and if you click cancel you should return to the project landing page after edit mode, and the projects list on my page after canceling "new project".

**Everything should now be good enough to remove the beta tag**. 
The actual removal will be done in a separate PR where I also remove all unused code. There are still a lot of open project tasks, but those are for features and improvements that also didn't exists in the old version. I will keep working on those once the beta tag is removed.

It doesn't look good on mobile when we have a table, but this is a problem everywhere as you know. The stepper is now removed on mobile, since we still have the next buttons.

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/pages/project/project_wizard/ProjectFormStepper.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const ProjectFormStepper = ({
     }
   };
 
-  return (
+  return isMobile ? null : (
     <Stepper nonLinear activeStep={tabNumber}>
```
Comment (elkebab) [inline src/pages/project/project_wizard/ProjectFormStepper.tsx:38]: Kunne også vært løst med ren CSS, sånn at det ikke vil trigge noen re-render. Ulempen(?) er at HTMLen fortsatt vil ligge i treet da, den bare skjules i CSS. Vet ikke hva som egentlig er beste practice av dem 
```suggestion
    <Stepper nonLinear activeStep={tabNumber} sx={{ display: { xs: 'none', md: 'flex' } }}>
```

File: src/pages/registration/description_tab/projects_field/ProjectsField.tsx [hunk 2]
```diff
@@ -128,6 +121,25 @@ export const ProjectsField = () => {
...
+                    </Button>
+                  ) : (
+                    <Button
+                      data-testid={dataTestId.registrationWizard.description.createProjectButton}
+                      onClick={() => setOpenNewProjectDialog(true)}
+                      startIcon={<AddIcon />}>
+                      {t('project.create_new_project')}
+                    </Button>
+                  )}
+                  {betaEnabled && <OpenInNewIcon />}
```
Comment (elkebab) [inline src/pages/registration/description_tab/projects_field/ProjectsField.tsx:141]: Skulle dette ikonet heller vært `endIcon` på knappen? Er vant til at ikonet er en del av linken, men det blir det ikke her. Ser kanskje litt rart ut med båre `startIcon` og `endIcon` også da men 😅

**Review by elkebab — APPROVED** [review]

File: src/pages/project/project_wizard/ProjectFormStepper.tsx (hunk 1 above)
Comment (denektenina) [reply src/pages/project/project_wizard/ProjectFormStepper.tsx:38]: Ja vet heller ikke. Dette er sånn det er gjort i den andre stepperen. Tror ikke det er noe problem at den ligger skjult i treet, men da blir den også rendret da.

File: src/pages/registration/description_tab/projects_field/ProjectsField.tsx (hunk 2 above)
Comment (denektenina) [reply src/pages/registration/description_tab/projects_field/ProjectsField.tsx:141]: Ja det var det jeg tenkte var litt rart. Sånn som det er nå blir det litt som spørsmålstegnet over, at det er mer en veiledning, men jeg vet ikke.

Comment (denektenina) [reply src/pages/registration/description_tab/projects_field/ProjectsField.tsx:141]: Spurte Mayli, så kan jeg endre senere om aktuelt

**Pull request merged**